Src/FluentAssertions/Common/DictionaryHelpers.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace FluentAssertions.Common
+{
+    internal static class DictionaryHelpers
+    {
+        public static IEnumerable<TKey> GetKeys<TCollection, TKey, TValue>(this TCollection collection)
+            where TCollection : IEnumerable<KeyValuePair<TKey, TValue>>
+        {
+            return collection switch
+            {
+                IDictionary<TKey, TValue> dictionary => dictionary.Keys,
+                IReadOnlyDictionary<TKey, TValue> readOnlyDictionary => readOnlyDictionary.Keys,
+                _ => collection.Select(kvp => kvp.Key),
+            };
+        }
+
+        public static IEnumerable<TValue> GetValues<TCollection, TKey, TValue>(this TCollection collection)
+            where TCollection : IEnumerable<KeyValuePair<TKey, TValue>>
+        {
+            return collection switch
```

Comment:
Although I don't like switching on types, it's a nice example of using the new C# syntax.

Reply:
I'm curious why :)
Would your normally prefer regular conditionals when needing to pattern match?

```c#
if (collection is IDictionary<TKey, TValue> dictionary)
{
   return dictionary.Values;
}

if (IReadOnlyDictionary<TKey, TValue> readOnlyDictionary)
{
   return readOnlyDictionary.Values;
}

return collection.Select(kvp => kvp.Value);
```

Reply:
For me, I think a possible reason could be that my eyes don't pick it up as a switch statement yet. They layout of the code is not implying it. But that might just take time.

Reply:
@eNeRGy164 as in the GH markdown formatting not coloring the keyword switch in the switch expression?
```c#
switch (a)
{
    case I i: return i;
}
```

```c#
return a switch
{
    I i => i;
}
```

Reply:
> I'm curious why :)

I prefer polymorphistic behavior ;-)